executor/metric_reader.go

```diff
@@ -62,11 +63,18 @@ func (e *MetricRetriever) retrieve(ctx context.Context, sctx sessionctx.Context)
 		quantiles = []float64{tblDef.Quantile}
 	}
 	for _, quantile := range quantiles {
-		queryValue, err := e.queryMetric(ctx, sctx, queryRange, quantile)
+		var queryValue pmodel.Value
+		// Add retry to avoid network error.
+		for i := 0; i < 10; i++ {
+			queryValue, err = e.queryMetric(ctx, sctx, queryRange, quantile)
+			if err == nil || strings.Contains(err.Error(), "parse error") {
```

Comment:
Can we be sure that all errors rather than "parse error" are retry-able?

Reply:
I can't enumerate all errors...
And retry doesn't affect correctness, retry is idempotent.